modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultUtils.java

```diff
@@ -12,40 +12,54 @@
 public class AssaultUtils {
 
     public static boolean areDefendersOnline(ProtectedRegion tRegion) {
-        HashSet<UUID> players = new HashSet<>();
-        players.addAll(tRegion.getOwners().getUniqueIds());
         int numOnline = 0;
 
         if(Settings.AssaultRequiredOwnersOnline > 0) {
-            for (UUID playerName : players) {
-                if (Bukkit.getPlayer(playerName) != null) {
+            HashSet<UUID> owners = new HashSet<>();
+            owners.addAll(tRegion.getOwners().getUniqueIds());
+            for (UUID playerID : owners) {
+                if (Bukkit.getPlayer(playerID) != null) {
                     numOnline++;
+
+                    if(numOnline > Settings.AssaultRequiredOwnersOnline) {
+                        break;
+                    }
                 }
             }
 
             if (numOnline < Settings.AssaultRequiredOwnersOnline) {
                 return false;
             }
-            numOnline = 0;
         }
 
-        players.addAll(tRegion.getMembers().getUniqueIds());
-        for (UUID playerName : players) {
-            if (Bukkit.getPlayer(playerName) != null) {
-                numOnline++;
+        numOnline = 0;
+        if(Settings.AssaultRequiredOwnersOnline > 0) {
+            HashSet<UUID> members = new HashSet<>();
+            members.addAll(tRegion.getMembers().getUniqueIds());
```

Comment:
This is again  unnecessary, just use `tRegion.getMembers().getUniqueIds()` for iteration